bundles/http/Citrix/Slack/2e8a194d-0c01-44c1-9a62-a5ee1d5ecbb3.js

```diff
@@ -129,48 +127,54 @@ async function changeIDbyRealName(dataStore, client, timestamp) {
 
         do {
 
-            let user_ids = []
-            let indexUser = 0;
-            let text = channelMessages[indexMessages].text;
-            let searchText = text;
+            let userIds = []
+            let userNames = []
+            let userRegex = /(?:^|\W)<@(\w+)(?!\w)/g;
+            let match;
+            let replaceResult = [];
 
-            do {
+            let text = channelMessages[indexMessages].text;
 
-                const user_id = searchText.substring(searchText.indexOf('<@') + 2, searchText.indexOf('>'));
+            while (match = userRegex.exec(text)) {
+                userIds.push(match[1]);
+            }
 
-                if (user_id) {
-                    user_ids.push(user_id)
-                } else {
-                    channelMessages[indexMessages].text_changed = text
-                }
+            channelMessages[indexMessages].text_changed = text
 
-                searchText = searchText.substring(searchText.indexOf('>') + 1)
+            // If there is an user tagged in the message
+            if (userIds.length) {
+                for (let j = 0; j < userIds.length; j++) {
+                    let userId = userIds[j]
+                    let resp = await client.fetch(`/users.info?user=${userId}`)
 
-                let resp = await client.fetch(`/users.info?user=${user_ids[indexUser]}`)
+                    if (!resp.ok) {
+                        throw new Error(`Could not retrieve user data (${resp.status}: ${resp.statusText})`);
+                    }
 
-                if (!resp.ok) {
-                    throw new Error(`Could not retrieve user data (${resp.status}: ${resp.statusText})`);
+                    let userData = await resp.json()
+                    if (userData.user === undefined) {
+                        indexChannels++
+                        break
```

Comment:
Are you sure that you want to break cycle?
imho here should be `continue` to check the rest of `userIds`